packages/@aws-cdk/aws-scheduler-alpha/lib/private/schedule.ts

```diff
@@ -36,19 +37,31 @@ export interface ScheduleProps {
    * @default - no value
    */
   readonly description?: string;
+
+  /**
+   * The schedule's group.
+   *
+   * @deafult - By default a schedule will be associated with the `default` group.
+   */
+  readonly group?: IGroup;
 }
 
 /**
  * An EventBridge Schedule
  */
 export class Schedule extends Resource implements ISchedule {
+  group?: IGroup;
+
   constructor(scope: Construct, id: string, props: ScheduleProps) {
     super(scope, id);
 
+    this.group = props.group;
```

Comment:
Not sure if we should do:

this.group = props.group ?? Group.fromGroupName(this, "DefaultGroup");

Reply:
```
Group.fromDefaultGroup(this, "DefaultGroup")
``` 

I think it shouldn't create a new resource, as we just refer to existing one by ARN? 

Why would we add a default group here? So customers can refer to it?

Reply:
Yeah, for:

```ts
const schedule = new Schedule(this, 'MySchedule', {
   /* anything but no group */
});
schedule.group.grantReadSchedules(role);
```

Reply:
Hm, interesting. I am not sure if group should be explicitly set by customer to `default` or not. 

Curious what @kaizencc and @comcalvi think about it? Do you have similar examples in the CDK library already?

Reply:
To answer the first question, calling `fromDefaultGroup` will not create a new resource so we are in the clear there.

I have a bit of an issue here, which is a scenario like this:

```ts
const s = new Schedule();
const dg = s.group // this would be the default group
const g = new Group();
g.addSchedules(s);

const ng = s.group;

expect(dg).toEqual(ng); // but this assertion will fail, which is unexpected
```

an additional question: can a schedule be a part of multiple groups? yes right?

```ts
const s = new Schedule();
const g1 = new Group();
g1.addSchedules(s);
const g2 = new Group();
g2.addSchedules(s);

// what will be the value for this?
s.group;
```

maybe the solution here is to expose a groups property and have the defaultGroup be an item in the list

Reply:
I think there is a confusion. Schedule can be placed only in one group.

> expect(dg).toEqual(ng); // but this assertion will fail, which is unexpected

I think it is expected to fail. Why do you think it is a problem?

Reply:
> Schedule can be placed only in one group.

Ok. Do we enforce that in the cdk? If so, I don't see a unit test for it.


> I think it is expected to fail. Why do you think it is a problem?

The problem is that `s.group` equals different things over the course of the cdk app. At one point it is the default group, at another it is the given group. We should try to avoid that. Maybe we can lazily evaluate `group` so that we know whether or not the group is the default one or provided later:

```ts
// when group.addSchedules(s); is called, we populate s._group

this.group = Lazy.produce(() => this._group ?? Group.fromDefaultGroup(this, "DefaultGroup"));
```

Reply:
> Ok. Do we enforce that in the cdk? If so, I don't see a unit test for it.

hm, this behaviour is enforced by CloudFormation.  But we could check that after two updates of the groups, the schedule has the latest group. 

> The problem is that s.group equals different things over the course of the cdk app 

Sorry, I am not following, why would the fact that the variable `group` is different through out app execution is a problem?

Reply:
1) If we know that a schedule can be placed only in one group, we should error out at synth time if we try to add a schedule to two groups. Should not wait for CFN behavior to kick in.

2) Because it is confusing. Someone may use the initial `group` to get metrics, or pass it to somewhere else in their cdk app, or assert its value, or whatever. Not knowing that they will override the `group` later in the CDK app. It's hard to reason about what is happening when a variable changes throughout app execution. Furthermore, I argue that it is _incorrect_ for `group` to be `defaultGroup` if later we add the schedule to a specific group.

Reply:
Maybe - to avoid confusion - we should not have the addSchedules.

The only way to set a group then is through the constructor. Which probably also makes the Lazy unnecessary.

Reply:
@kaizencc, thank you! Your arguments make sense to me. I originally thought that mutable variables should not be an issue for CDK as we use an imperative language, but CDK classes are declarative. Plus in general, immutable is less confusing than mutable. 

I like Jacco's suggestion. I will remove method `addSchedules`. I like that there will be only a single way to put `Schedule` in a `Group` (through `Schedule` properties).

Reply:
We can always worry about adding in `addSchedules` later, if people want